wrappers/systemd-snmp.service.in

```diff
@@ -11,6 +11,18 @@ After=network.target
 Type=forking
 PIDFile=/run/dirsrv/ldap-agent.pid
 ExecStart=@sbindir@/ldap-agent @configdir@/ldap-agent.conf
+PrivateTmp=on
+# https://en.opensuse.org/openSUSE:Security_Features#Systemd_hardening_effort
+ProtectSystem=full
+ProtectHome=true
+PrivateDevices=true
+ProtectHostname=true
```

Comment:
RHEL8 has systemd-239 which doesn't support some of these options (`ProtectHostname` was added in 242, `ProtectKernelLogs` in 244 and `ProtectClock` in 245):
```
Nov 23 08:32:28 rhel85 systemd[1]: /usr/lib/systemd/system/dirsrv@.service.d/custom.conf:57: Unknown lvalue 'ProtectHostname' in section 'Service'
Nov 23 08:32:28 rhel85 systemd[1]: /usr/lib/systemd/system/dirsrv@.service.d/custom.conf:58: Unknown lvalue 'ProtectClock' in section 'Service'
Nov 23 08:32:28 rhel85 systemd[1]: /usr/lib/systemd/system/dirsrv@.service.d/custom.conf:61: Unknown lvalue 'ProtectKernelLogs' in section 'Service'
```

The page you linked has an rpm macro to remove these options if systemd is too old. Could you please also add it to our spec file?

And since we're adding these to the main systemd unit file, we can remove these from the drop-in config:
https://github.com/389ds/389-ds-base/blob/e2839d51b7f92f8437d3ffd0e1c9a35ef7ed250b/wrappers/systemd.template.service.custom.conf.in#L46-L50

Reply:
Thanks mate, applied both suggestions!